amazon-kinesis-client/src/main/java/software/amazon/kinesis/common/ConfigsBuilder.java

```diff
@@ -108,6 +109,8 @@ public String namespace() {
         return namespace;
     }
 
+    private MultiStreamTracker multiStreamTracker;
```

Comment:
Why being optional is a blocker for being a final/immutable? 
Let's model as Optional<MultistreamTracker>, and set to absent for cases where it's not used.

Reply:
We use lombok builder for building this config with mandatory fields like streamName, client passed in the constructor and optional fields can be set through the builder. If we make tracker as immutable, then we have to initialize the tracker in constructor with Optional.empty(), which will prevent the overriding through the builder. I preferred to introduce any new config as optional, to avoid creating explicit constructors. Alternatively, we can explicitly add new constructors that either allows to set streamName or MultiStreamTracker.